pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -100,6 +101,18 @@ func matchesIncomingPodTerm(pod *v1.Pod, termLabelSelector labels.Selector, term
 	return termLabelSelector.Matches(labels.Set(pod.Labels))
 }
 
+// GetNamespaceLabelsSnapshot returns a snapshot of the labels associated with
+// the namespace.
+func GetNamespaceLabelsSnapshot(ns string, nsLister listersv1.NamespaceLister) (nsLabels labels.Set) {
+	podNS, err := nsLister.Get(ns)
+	if err == nil {
+		// Create and return snapshot of the labels.
+		return labels.Merge(podNS.Labels, nil)
```

Comment:
I still have the same question, do we really need to copy? The lister gives us a copy already, doesn't it?

Reply:
I thought it gives us a reference to the underlying cache object.

Reply:
You were right, it's the same object 😓